### PR TITLE
pipelinehelper better exception handling

### PR DESCRIPTION
**Pull request by Ginxo**

### CODE REVIEW

File: vars/PipelineHelper.groovy [hunk 1]
```diff
@@ -7,23 +8,27 @@ def class PipelineHelper {
...
-                    steps.println "[ERROR] Timeout exceeded in ${counter + 1}/${maxAttempts}"
-                    steps.error("Failing build because Timeout ${counter + 1}/${maxAttempts}")
-                    if(errorAction) {
-                        try {
-                            errorAction.call();
-                        } catch(Exception errorActionException) {
-                            steps.println "[ERROR] Error executing error action ${errorActionException.getMessage()}"
-                            throw errorActionException
+                    action.call()
+                } catch (Exception | FlowInterruptedException e) {
```
Comment (radtriste) [inline vars/PipelineHelper.groovy:18]: I guess that can be changed to 
```suggestion
                } catch (Exception e) {
```

**Review by radtriste — APPROVED** [review]

**Comment by Ginxo** [comment]
can't test the random error but the job still working https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/drools-master.pr/400/console

**Comment by radtriste** [comment]
please add a simple stage to `.jenkins/Jenkinsfile` that at least run the retry command to check everything we use there is supported by current Jenkins

**Review by mbiarnes — APPROVED** [review]
LGTM

**Pull request merged**